graphql/src/schema/api.rs

```diff
@@ -27,9 +27,9 @@ pub fn api_schema(input_schema: &Document) -> Result<Document, APISchemaError> {
     let mut schema = input_schema.clone();
     add_builtin_scalar_types(&mut schema)?;
     add_order_direction_enum(&mut schema);
-    add_field_arguments(&mut schema, &input_schema)?;
     add_types_for_object_types(&mut schema, &object_types)?;
     add_types_for_interface_types(&mut schema, &interface_types)?;
+    add_field_arguments(&mut schema, &input_schema)?;
```

Comment:
This may be worth a comment about why it's important to do this after adding the input object types first. But since we're adding tests here and they will break if anyone messes with this, I'm fine without a comment.

Reply:
Curious though: How did this even work for top-level fields?

Reply:
That is in `add_query_type` which is done after. Yhea I added the test because I was too lazy to write a comment 😛